# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsAccUtils.cpp
# Commit: 198145495664
# Full Hash: 19814549566419d05996f0d8811d8216cce40308
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-11-10 09:23:09
# Regressor Bug: 1796733
# File Overlap Count: 1
# Description:
#   Bug 1796733 part 5: Add an overload of nsAccUtils::NormalizeARIAToken which takes an Element. r=avandolder
#   
#   The next patch in this stack adds a method to retrieve aria-selected.
#   The easiest way to do this is using NormalizeARIAToken.
#   This overload takes an Element and checks both the Element and ElementInternals.
# ==============================================================================

diff -r d07307df1220 -r 198145495664 accessible/base/nsAccUtils.cpp
--- a/accessible/base/nsAccUtils.cpp	Thu Nov 10 04:19:27 2022 +0000
+++ b/accessible/base/nsAccUtils.cpp	Thu Nov 10 04:19:27 2022 +0000
@@ -176,6 +176,19 @@
   return nullptr;
 }
 
+nsStaticAtom* nsAccUtils::NormalizeARIAToken(dom::Element* aElement,
+                                             nsAtom* aAttr) {
+  if (auto* htmlElement = nsGenericHTMLElement::FromNode(aElement);
+      htmlElement && !aElement->HasAttr(aAttr)) {
+    const auto* defaults = GetARIADefaults(htmlElement);
+    if (!defaults) {
+      return nsGkAtoms::_empty;
+    }
+    return NormalizeARIAToken(defaults, aAttr);
+  }
+  return NormalizeARIAToken(&aElement->GetAttrs(), aAttr);
+}
+
 Accessible* nsAccUtils::GetSelectableContainer(const Accessible* aAccessible,
                                                uint64_t aState) {
   if (!aAccessible) return nullptr;